src/Operation/BulkWrite.php

```diff
@@ -330,13 +341,31 @@ public function execute(Server $server)
                     break;
 
                 case self::INSERT_ONE:
-                    $insertedIds[$i] = $bulk->insert($args[0]);
+                    $insertedDocument = isset($this->options['codec'])
+                        ? $this->options['codec']->encodeIfSupported($args[0])
+                        : $args[0];
+                    // Psalm's assert-if-true annotation does not work with unions, so
+                    // assert the type manually instead of using is_document
```

Comment:
Is there an upstream bug report we want to reference here?

Reply:
Added a reference to https://github.com/vimeo/psalm/issues/6831